app/models/vm_scan.rb

```diff
@@ -43,32 +43,33 @@ def call_check_policy
     begin
       vm = VmOrTemplate.find(target_id)
 
-      cb = {
-        :class_name  => self.class.to_s,
-        :instance_id => id,
-        :method_name => :check_policy_complete,
-        :server_guid => MiqServer.my_guid
+      q_options = {
+        :miq_callback => {
+          :class_name  => self.class.to_s,
+          :instance_id => id,
+          :method_name => :check_policy_complete,
+          :args        => [MiqServer.my_zone] # store the zone where the scan event is created
+        }
       }
       inputs = {:vm => vm, :host => vm.host}
-      if !MiqEvent.raise_evm_job_event(vm, {:type => "scan", :suffix => "start"}, inputs, :miq_callback => cb)
-        msg = "Aborted policy resolution - scan event was not raised to automate."
-        _log.error(msg)
-        signal(:abort, msg, "error")
-      end
+      MiqEvent.raise_evm_job_event(vm, {:type => "scan", :suffix => "start"}, inputs, q_options)
     rescue => err
       _log.log_backtrace(err)
       signal(:abort, err.message, "error")
-    rescue Timeout::Error
-      msg = "Request to check policy timed out"
-      _log.error(msg)
-      signal(:abort, msg, "error")
     end
   end
 
-  def check_policy_complete(status, message, result)
+  def check_policy_complete(from_zone, status, message, result)
     unless status == 'ok'
       _log.error("Status = #{status}, message = #{message}")
-      signal(:abort, message, "error")
+      MiqQueue.put_unless_exists(
+        :class_name  => self.class.to_s,
+        :instance_id => id,
+        :method_name => "signal",
+        :args        => [:abort, message, "error"],
+        :zone        => from_zone, # continue in the same zone where the scan event is created
```

Comment:
Let's update the comment to say: `# Continue in the same zone were the scan job was initiated.`